src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,78 @@
-export const Pagination = () => {};
+import React from 'react';
+import { getNumbers } from '../../utils';
+import cn from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (val: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pages = getNumbers(1, Math.ceil(total / perPage));
+  const handlePrev = () => {
+    if (currentPage === 1) {
+      return;
+    }
+
+    onPageChange(currentPage - 1);
+  };
+
+  const handleNext = () => {
+    if (currentPage === pages.length) {
+      return;
+    }
+
+    onPageChange(currentPage + 1);
+  };
+
+  return (
+    <ul className="pagination">
+      <li className={cn('page-item', { disabled: currentPage === 1 })}>
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={handlePrev}
+        >
+          «
+        </a>
+      </li>
+      {pages.map(num => (
+        <li
+          key={num}
+          className={cn('page-item', { active: currentPage === num })}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${num}`}
+            onClick={() => onPageChange(num)}
```

Comment:
Similarly, for the page links, you should prevent the default action by calling `event.preventDefault()` in the `onClick` handler to avoid changing the URL fragment identifier and potential page jumps.